server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala

```diff
@@ -0,0 +1,108 @@
+ /***************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.api.upload
+
+import org.apache.commons.io.IOUtils
+import org.apache.james.core.Username
+import org.apache.james.jmap.api.exception.UploadNotFoundException
+import org.apache.james.jmap.api.model.{Upload, UploadId}
+import org.apache.james.jmap.api.upload.UploadRepositoryContract.{CONTENT_TYPE, DATA, DATA_STRING, USER}
+import org.apache.james.mailbox.model.ContentType
+import org.assertj.core.api.Assertions.{assertThat, assertThatThrownBy}
+import org.junit.jupiter.api.Test
+import reactor.core.scala.publisher.SMono
+
+import java.io.InputStream
+import java.nio.charset.StandardCharsets
+
+
+object UploadRepositoryContract {
+  private lazy val CONTENT_TYPE: ContentType = ContentType
+    .of("text/html")
+  private lazy val DATA_STRING: String = "123321"
+  private lazy val DATA: InputStream = IOUtils.toInputStream(DATA_STRING, StandardCharsets.UTF_8)
+  private lazy val USER: Username = Username.of("Bob")
+}
+
+
+trait UploadRepositoryContract {
+
+  def testee: UploadRepository
+
+  @Test
+  def uploadShouldSuccess(): Unit = {
+    val uploadId: UploadId = SMono.fromPublisher(testee.upload(DATA, CONTENT_TYPE, USER)).block()
+
+    assertThat(SMono.fromPublisher(testee.retrieve(uploadId, USER)).block())
+      .isNotNull
+  }
+
+  @Test
+  def uploadShouldReturnSameIdWhenReUploadSameContent(): Unit = {
```

Comment:
IMO not needed. Dedup should be a blob store concern...

Reply:
What about ADR? 
(https://github.com/apache/james-project/pull/544/files#L22)
```
if reuploaded, the same blobId MAY be returned, but this SHOULD reset the expiry time.
```

Reply:
Idem